proj1a/ArrayDeque.java

```diff
@@ -0,0 +1,185 @@
+public class ArrayDeque<T> implements Deque<T> {
+    private T[] items;
+    private int size;
+    private int nextFirst;
+    private int nextLast;
+    private int RFACTOR;
+    private int R;
+
+
+    /**create empty array deque*/
+    public ArrayDeque() {
+        items = (T[]) new Object[8];
+        size = 0;
+        nextFirst = 4;
+        nextLast = 5;
+    }
+
+    /**calculate nextFirst*/
+    private int nextF(int i) {
+        int next;
+        if (i == 0) {
+            next = items.length - 1;
+        } else {
+            next = i - 1;
+        }
+        return next;
+    }
+
+    /**calculate nextLast*/
+    private int nextL(int i) {
+        int next;
+        if (i == items.length - 1) {
+            next = 0;
+        } else {
+            next = i + 1;
+        }
+        return next;
+    }
+
+    /**resize array, create a larger one*/
+    private void resizeL(int capacity) {
+        T[] temp = (T[]) new Object[capacity];
+        int old_size = size;
+        size = 0;
+        nextFirst = 4;
+        nextLast = 5;
+        for (int i = 0; i < old_size ; i++ ) {
+            addFirst((T) get(i));
```

Comment:
How does this line of code add items to the `temp` array? Where is the code unit that unit tests this function?